In [34]:
cd /content/drive/MyDrive/Not_Synced

/content/drive/MyDrive/Not_Synced


In [38]:
ls

audio_hparams.py    logs/                    readme.gdoc
audio.py            lrs2_v1.tar              README.md
batch_inference.py  part_files_DELETE/       requirements.txt
discriminator.py    preprocessed_files/      results/
filelists/          preprocess.py            train.py
generator.py        processed_folder.tar.gz  train_unet.py
License.md          __pycache__/


In [36]:
ls filelists/

pretrain.txt  README.md  test.txt  train.txt  val.txt


In [39]:
ls logs

gen_faces/  README.md


Prerequisites
-------------
- Python >= 3.5
- ffmpeg: `sudo apt-get install ffmpeg`
- Install necessary packages using `pip install -r requirements.txt`
- Install keras-contrib `pip install git+https://www.github.com/keras-team/keras-contrib.git`
- git clone `https://github.com/Rudrabha/LipGAN.git` (make sure to check out to branch `fully_pythonic`)

Training LipGAN
-------
We illustrate the training pipeline using the LRS2 dataset. Adapting for other datasets would involve small modifications to the code. 

###### LRS2 dataset folder structure
```
data_root (mvlrs_v1)
├── main, pretrain (we use only main folder in this work)
|	├── list of folders
|	│   ├── five-digit numbered video IDs ending with (.mp4)
```


In [15]:
cp -r /content/drive/MyDrive/Not_Synced/processed_folder.tar.gz /content/LipGAN/

^C


In [ ]:
cd /content/LipGAN/

/content/LipGAN


In [ ]:
ls

batch_inference.py  License.md     processed_folder.tar.gz  train.py
discriminator.py    logs/          README.md                train_unet.py
filelists/          matlab/        requirements.txt
generator.py        preprocess.py  results/


In [ ]:
!tar -xvf processed_folder.tar.gz

Streaming output truncated to the last 5000 lines.
content/preprocessed_files/main/6117976902660304352/00016/18.jpg
content/preprocessed_files/main/6117976902660304352/00016/0.jpg
content/preprocessed_files/main/6117976902660304352/00007/
content/preprocessed_files/main/6117976902660304352/00007/mels.npz
content/preprocessed_files/main/6117976902660304352/00007/15.jpg
content/preprocessed_files/main/6117976902660304352/00007/25.jpg
content/preprocessed_files/main/6117976902660304352/00007/14.jpg
content/preprocessed_files/main/6117976902660304352/00007/11.jpg
content/preprocessed_files/main/6117976902660304352/00007/20.jpg
content/preprocessed_files/main/6117976902660304352/00007/19.jpg
content/preprocessed_files/main/6117976902660304352/00007/audio.wav
content/preprocessed_files/main/6117976902660304352/00007/10.jpg
content/preprocessed_files/main/6117976902660304352/00007/17.jpg
content/preprocessed_files/main/6117976902660304352/00007/9.jpg
content/preprocessed_files/main/6117976902

In [ ]:
ls

batch_inference.py  generator.py  preprocess.py            results/
content/            License.md    processed_folder.tar.gz  train.py
discriminator.py    logs/         README.md                train_unet.py
filelists/          matlab/       requirements.txt


In [ ]:
mv content/preprocessed_files .

In [ ]:
ls

batch_inference.py  generator.py  preprocessed_files/      requirements.txt
content/            License.md    preprocess.py            results/
discriminator.py    logs/         processed_folder.tar.gz  train.py
filelists/          matlab/       README.md                train_unet.py


In [ ]:
rm -rf content/

In [ ]:
ls

batch_inference.py  License.md           preprocess.py            results/
discriminator.py    logs/                processed_folder.tar.gz  train.py
filelists/          matlab/              README.md                train_unet.py
generator.py        preprocessed_files/  requirements.txt


In [ ]:
ls preprocessed_files/

main/


In [14]:
#!pip install -r requirements.txt

In [1]:
pip install Keras==2.2.4 Pillow==5.4.1 bleach==1.5.0 dlib==19.17.0 h5py==2.9.0 html5lib==0.9999999 keras-applications==1.0.8 librosa==0.7.0 matplotlib==2.1.0  opencv-python==4.0.0.21 scikit-image==0.15.0  soundfile==0.10.3.post1 tensorboard==1.14.0 tensorflow-gpu==1.14.0 tqdm==4.19.9


     |████████████████████████████████| 317kB 6.4MB/s 
     |████████████████████████████████| 2.0MB 8.3MB/s 
     |████████████████████████████████| 3.4MB 22.2MB/s 
     |████████████████████████████████| 2.8MB 50.5MB/s 
     |████████████████████████████████| 890kB 54.1MB/s 
     |████████████████████████████████| 51kB 6.0MB/s 
     |████████████████████████████████| 1.6MB 56.1MB/s 
     |████████████████████████████████| 35.7MB 198kB/s 
     |████████████████████████████████| 25.4MB 1.2MB/s 
     |████████████████████████████████| 26.3MB 1.2MB/s 
     |████████████████████████████████| 3.2MB 60.2MB/s 
     |████████████████████████████████| 377.1MB 26kB/s 
     |████████████████████████████████| 61kB 6.9MB/s 
     |████████████████████████████████| 491kB 60.8MB/s 
  Created wheel for dlib: filename=dlib-19.17.0-cp37-cp37m-linux_x86_64.whl size=3866546 sha256=6e760e0bfa1e14f19bf3aa62628c4690889804fc2e260cf4f662ecaddddb77bd
  Stored in directory: /root/.cache/pip/wheels/50/b6/b5/5f46a

In [41]:
pip install  numpy==1.17 #scipy==1.0.0

     |████████████████████████████████| 20.3MB 64.7MB/s 
ERROR: tensorflow-probability 0.12.1 has requirement gast>=0.3.2, but you'll have gast 0.2.2 which is incompatible.
ERROR: datascience 0.10.6 has requirement folium==0.2.1, but you'll have folium 0.8.3 which is incompatible.
ERROR: albumentations 0.1.12 has requirement imgaug<0.2.7,>=0.2.5, but you'll have imgaug 0.2.9 which is incompatible.
  Found existing installation: numpy 1.19.5
    Uninstalling numpy-1.19.5:
      Successfully uninstalled numpy-1.19.5


In [1]:
!pip uninstall -y tensorflow
!pip install tensorflow

Uninstalling tensorflow-2.4.1:
  Successfully uninstalled tensorflow-2.4.1
     |████████████████████████████████| 394.3MB 35kB/s 


In [2]:
pip install numba==0.48

     |████████████████████████████████| 3.6MB 6.7MB/s 
     |████████████████████████████████| 20.2MB 490kB/s 
  Found existing installation: llvmlite 0.34.0
    Uninstalling llvmlite-0.34.0:
      Successfully uninstalled llvmlite-0.34.0
  Found existing installation: numba 0.51.2
    Uninstalling numba-0.51.2:
      Successfully uninstalled numba-0.51.2


In [3]:
!pip install git+https://www.github.com/keras-team/keras-contrib.git


  Cloning https://www.github.com/keras-team/keras-contrib.git to /tmp/pip-req-build-_dafoic2
  Running command git clone -q https://www.github.com/keras-team/keras-contrib.git /tmp/pip-req-build-_dafoic2
  Created wheel for keras-contrib: filename=keras_contrib-2.0.8-cp37-none-any.whl size=101065 sha256=c98d2899d2ea85d3d8c07f399d96830b578eb22d80cdc6b40e29027d781fb092
  Stored in directory: /tmp/pip-ephem-wheel-cache-9o3ninu9/wheels/11/27/c8/4ed56de7b55f4f61244e2dc6ef3cdbaff2692527a2ce6502ba
Successfully built keras-contrib


In [4]:
pip install tensorflow-gpu==1.15.2

     |████████████████████████████████| 411.0MB 33kB/s 
     |████████████████████████████████| 3.8MB 56.2MB/s 
     |████████████████████████████████| 512kB 51.9MB/s 
  Created wheel for gast: filename=gast-0.2.2-cp37-none-any.whl size=7540 sha256=c7d953db07ae818f84d4755035fe8c464f7804362de5ff0a0eefc5a6a55dd670
  Stored in directory: /root/.cache/pip/wheels/5c/2e/7e/a1d4d4fcebe6c381f378ce7743a3ced3699feb89bcfbdadadd
Successfully built gast
ERROR: tensorflow 2.4.1 has requirement gast==0.3.3, but you'll have gast 0.2.2 which is incompatible.
ERROR: tensorflow 2.4.1 has requirement h5py~=2.10.0, but you'll have h5py 2.9.0 which is incompatible.
ERROR: tensorflow 2.4.1 has requirement tensorboard~=2.4, but you'll have tensorboard 1.15.0 which is incompatible.
ERROR: tensorflow 2.4.1 has requirement tensorflow-estimator<2.5.0,>=2.4.0, but you'll have tensorflow-estimator 1.15.1 which is incompatible.
ERROR: tensorflow-probability 0.12.1 has requirement gast>=0.3.2, but you'll have gast 0.

In [5]:
cp -r /content/drive/MyDrive/LipGan1/LipGAN_pythonic /content/drive/MyDrive/Not_Synced

In [6]:
cp -r /content/drive/MyDrive/Not_Synced/processed_folder.tar.gz /content

In [7]:
!tar -xf processed_folder.tar.gz 

In [33]:
cd /content/drive/MyDrive/Not_Synced/LipGAN_pythonic

/content/drive/MyDrive/Not_Synced/LipGAN_pythonic


In [34]:
ls

audio_hparams.py    filelists/    preprocess.py     train.py
audio.py            generator.py  README.md         train_unet.py
batch_inference.py  License.md    requirements.txt
discriminator.py    logs/         results/


In [35]:
pwd

'/content/drive/MyDrive/Not_Synced/LipGAN_pythonic'

In [36]:
ls logs/

gen_faces/  README.md


In [37]:
rm logs/filenames.pkl

rm: cannot remove 'logs/filenames.pkl': No such file or directory


In [ ]:
!python train.py --data_root /content/content/preprocessed_files

Using TensorFlow backend.
Will be training on 1677860 images






2021-04-29 23:45:39.805091: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2021-04-29 23:45:39.811697: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2299995000 Hz
2021-04-29 23:45:39.814464: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5587bb9208c0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-04-29 23:45:39.814506: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2021-04-29 23:45:39.816598: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2021-04-29 23:45:39.826698: E tensorflow/stream_executor/cuda/cuda_driver.cc:318] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2021-04-29 23:45:39

In [21]:
ls logs

gen_faces/  README.md


In [26]:
cp -r /content/drive/MyDrive/LipGan1/LipGAN_pythonic /content/

In [27]:
cd /content/LipGAN_pythonic/

/content/LipGAN_pythonic


In [30]:
rm logs/filenames.pkl

In [31]:
!python train.py --data_root /content/content/preprocessed_files

Using TensorFlow backend.
Will be training on 1677860 images






2021-04-29 23:44:31.216077: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2021-04-29 23:44:31.222602: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2299995000 Hz
2021-04-29 23:44:31.225674: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5643c1b56e00 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-04-29 23:44:31.225753: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2021-04-29 23:44:31.227916: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2021-04-29 23:44:31.238167: E tensorflow/stream_executor/cuda/cuda_driver.cc:318] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
2021-04-29 23:44:31

In [ ]:
cd /content/drive/MyDrive/LipGan1/LipGAN_pythonic

In [ ]:
cd /content/LipGAN/

/content/LipGAN


In [ ]:
ls

batch_inference.py  matlab/                  requirements.txt
discriminator.py    preprocessed_files/      results/
filelists/          preprocess.py            train.py
generator.py        processed_folder.tar.gz  train_unet.py
License.md          __pycache__/
logs/               README.md


In [5]:
!git clone https://github.com/Rudrabha/LipGAN.git

Cloning into 'LipGAN'...
remote: Enumerating objects: 202, done.
remote: Counting objects: 100% (202/202), done.
remote: Compressing objects: 100% (145/145), done.
remote: Total 202 (delta 117), reused 101 (delta 51), pack-reused 0
Receiving objects: 100% (202/202), 492.95 KiB | 3.52 MiB/s, done.
Resolving deltas: 100% (117/117), done.


In [ ]:
%ls

batch_inference.py  matlab/                  requirements.txt
discriminator.py    preprocessed_files/      results/
filelists/          preprocess.py            train.py
generator.py        processed_folder.tar.gz  train_unet.py
License.md          __pycache__/
LipGAN/             README.md


In [7]:
mv LipGAN/* .

In [ ]:
pip list

In [ ]:
# pip install numpy==1.14

Will be training on 1677860 


In [12]:
pip install keras==2.1.2

     |████████████████████████████████| 307kB 4.2MB/s 
ERROR: textgenrnn 1.4.1 has requirement keras>=2.1.5, but you'll have keras 2.1.2 which is incompatible.
  Found existing installation: Keras 2.4.3
    Uninstalling Keras-2.4.3:
      Successfully uninstalled Keras-2.4.3


COPY everything to VM

In [28]:
cp -r /content/drive/MyDrive/Not_Synced/processed_folder.tar.gz /content

In [29]:
ls /content/LipGAN/

batch_inference.py  generator.py  preprocess.py  train.py
discriminator.py    logs/         results/       train_unet.py
filelists/          matlab/       sample_data/


In [35]:
pwd

'/content'

In [36]:
!tar -xf processed_folder.tar.gz 

In [1]:
cd /content/LipGAN/

/content/LipGAN


In [2]:
ls

batch_inference.py  generator.py  preprocess.py  sample_data/
discriminator.py    logs/         __pycache__/   train.py
filelists/          matlab/       results/       train_unet.py


In [ ]:
!python train_unet.py --data_root /content/content/preprocessed_files #--model 'basic'

In [3]:
pip list

Package                       Version       
----------------------------- --------------
absl-py                       0.12.0        
alabaster                     0.7.12        
albumentations                0.1.12        
altair                        4.1.0         
appdirs                       1.4.4         
argon2-cffi                   20.1.0        
astor                         0.8.1         
astropy                       4.2.1         
astunparse                    1.6.3         
async-generator               1.10          
atari-py                      0.2.6         
atomicwrites                  1.4.0         
attrs                         20.3.0        
audioread                     2.1.9         
autograd                      1.3           
Babel                         2.9.0         
backcall                      0.2.0         
beautifulsoup4                4.6.3         
bleach                        3.3.0         
blis                          0.4.1         
bokeh     

In [ ]:
ls preprocessed_files/main/5970667684845379863/00004

0.jpg   12.jpg  15.jpg  18.jpg  2.jpg  5.jpg  8.jpg      mels.npz
10.jpg  13.jpg  16.jpg  19.jpg  3.jpg  6.jpg  9.jpg
11.jpg  14.jpg  17.jpg  1.jpg   4.jpg  7.jpg  audio.wav


In [ ]:
# !python train_unet.py --data_root /content/LipGAN/preprocessed_files #--model 'basic'

In [26]:
ls /content/content/preprocessed_files/main/5535415699068794046/00001

0.jpg   12.jpg  15.jpg  18.jpg  20.jpg  23.jpg  26.jpg  3.jpg  6.jpg  9.jpg
10.jpg  13.jpg  16.jpg  19.jpg  21.jpg  24.jpg  27.jpg  4.jpg  7.jpg  audio.wav
11.jpg  14.jpg  17.jpg  1.jpg   22.jpg  25.jpg  2.jpg   5.jpg  8.jpg  mels.npz


In [27]:
cp -r /content/drive/MyDrive/LipGan1/LipGAN_pythonic .

In [3]:
cd /content/LipGAN_pythonic/

/content/LipGAN_pythonic


In [ ]:
!python train_unet.py --data_root /content/content/preprocessed_files

Using TensorFlow backend.
Will be training on 1677860 images



__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
input_face (InputLayer)         (None, 96, 96, 6)    0                                            
__________________________________________________________________________________________________
conv2d_1 (Conv2D)               (None, 96, 96, 32)   9440        input_face[0][0]                 
__________________________________________________________________________________________________
batch_normalization_1 (BatchNor (None, 96, 96, 32)   128         conv2d_1[0][0]                   
__________________________________________________________________________________________________
activation_1 (Activation)       (None, 96, 96, 32)   0           batch_normalization_1[0][0]      
_____________________________________________

In [1]:
import tensorflow as tf
device_name = tf.test.gpu_device_name()
# if device_name != '/device:GPU:0':
#   raise SystemError('GPU device not found')
print('Found GPU at: {}'.format(device_name))

Found GPU at: 


In [2]:
%tensorflow_version 2.x
import tensorflow as tf
import timeit

device_name = tf.test.gpu_device_name()
if device_name != '/device:GPU:0':
  print(
      '\n\nThis error most likely means that this notebook is not '
      'configured to use a GPU.  Change this in Notebook Settings via the '
      'command palette (cmd/ctrl-shift-P) or the Edit menu.\n\n')
  raise SystemError('GPU device not found')

def cpu():
  with tf.device('/cpu:0'):
    random_image_cpu = tf.random.normal((100, 100, 100, 3))
    net_cpu = tf.keras.layers.Conv2D(32, 7)(random_image_cpu)
    return tf.math.reduce_sum(net_cpu)

def gpu():
  with tf.device('/device:GPU:0'):
    random_image_gpu = tf.random.normal((100, 100, 100, 3))
    net_gpu = tf.keras.layers.Conv2D(32, 7)(random_image_gpu)
    return tf.math.reduce_sum(net_gpu)
  
# We run each op once to warm up; see: https://stackoverflow.com/a/45067900
cpu()
gpu()

# Run the op several times.
print('Time (s) to convolve 32x7x7x3 filter over random 100x100x100x3 images '
      '(batch x height x width x channel). Sum of ten runs.')
print('CPU (s):')
cpu_time = timeit.timeit('cpu()', number=10, setup="from __main__ import cpu")
print(cpu_time)
print('GPU (s):')
gpu_time = timeit.timeit('gpu()', number=10, setup="from __main__ import gpu")
print(gpu_time)
print('GPU speedup over CPU: {}x'.format(int(cpu_time/gpu_time)))



This error most likely means that this notebook is not configured to use a GPU.  Change this in Notebook Settings via the command palette (cmd/ctrl-shift-P) or the Edit menu.




SystemError: ignored

Using TensorFlow backend.
Will be training on 1677860 images
WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:517: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:74: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:4138: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:174: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:181: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:186: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

2021-04-28 05:44:47.251734: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2021-04-28 05:44:47.255773: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2000165000 Hz
2021-04-28 05:44:47.257252: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x564fe0f5c8c0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-04-28 05:44:47.257293: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:190: The name tf.global_variables is deprecated. Please use tf.compat.v1.global_variables instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:199: The name tf.is_variable_initialized is deprecated. Please use tf.compat.v1.is_variable_initialized instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:206: The name tf.variables_initializer is deprecated. Please use tf.compat.v1.variables_initializer instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:1834: The name tf.nn.fused_batch_norm is deprecated. Please use tf.compat.v1.nn.fused_batch_norm instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:133: The name tf.placeholder_with_default is deprecated. Please use tf.compat.v1.placeholder_with_default instead.

__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_face (InputLayer)         (None, 96, 96, 6)    0                                            
__________________________________________________________________________________________________
conv2d_1 (Conv2D)               (None, 96, 96, 32)   9440        input_face[0][0]                 
__________________________________________________________________________________________________
batch_normalization_1 (BatchNor (None, 96, 96, 32)   128         conv2d_1[0][0]                   
__________________________________________________________________________________________________
activation_1 (Activation)       (None, 96, 96, 32)   0           batch_normalization_1[0][0]      
__________________________________________________________________________________________________
conv2d_2 (Conv2D)               (None, 48, 48, 64)   51264       activation_1[0][0]               
__________________________________________________________________________________________________
batch_normalization_2 (BatchNor (None, 48, 48, 64)   256         conv2d_2[0][0]                   
__________________________________________________________________________________________________
activation_2 (Activation)       (None, 48, 48, 64)   0           batch_normalization_2[0][0]      
__________________________________________________________________________________________________
conv2d_3 (Conv2D)               (None, 48, 48, 64)   36928       activation_2[0][0]               
__________________________________________________________________________________________________
batch_normalization_3 (BatchNor (None, 48, 48, 64)   256         conv2d_3[0][0]                   
__________________________________________________________________________________________________
activation_3 (Activation)       (None, 48, 48, 64)   0           batch_normalization_3[0][0]      
__________________________________________________________________________________________________
conv2d_4 (Conv2D)               (None, 48, 48, 64)   36928       activation_3[0][0]               
__________________________________________________________________________________________________
batch_normalization_4 (BatchNor (None, 48, 48, 64)   256         conv2d_4[0][0]                   
__________________________________________________________________________________________________
activation_4 (Activation)       (None, 48, 48, 64)   0           batch_normalization_4[0][0]      
__________________________________________________________________________________________________
add_1 (Add)                     (None, 48, 48, 64)   0           activation_4[0][0]               
                                                                 activation_2[0][0]               
__________________________________________________________________________________________________
input_audio (InputLayer)        (None, 12, 35, 1)    0                                            
__________________________________________________________________________________________________
activation_5 (Activation)       (None, 48, 48, 64)   0           add_1[0][0]                      
__________________________________________________________________________________________________
conv2d_27 (Conv2D)              (None, 12, 35, 128)  1280        input_audio[0][0]                
__________________________________________________________________________________________________
conv2d_5 (Conv2D)               (None, 48, 48, 64)   36928       activation_5[0][0]               
__________________________________________________________________________________________________
batch_normalization_27 (BatchNo (None, 12, 35, 128)  512         conv2d_27[0][0]                  
__________________________________________________________________________________________________
batch_normalization_5 (BatchNor (None, 48, 48, 64)   256         conv2d_5[0][0]                   
__________________________________________________________________________________________________
activation_36 (Activation)      (None, 12, 35, 128)  0           batch_normalization_27[0][0]     
__________________________________________________________________________________________________
activation_6 (Activation)       (None, 48, 48, 64)   0           batch_normalization_5[0][0]      
__________________________________________________________________________________________________
conv2d_28 (Conv2D)              (None, 12, 35, 128)  147584      activation_36[0][0]              
__________________________________________________________________________________________________
conv2d_6 (Conv2D)               (None, 48, 48, 64)   36928       activation_6[0][0]               
__________________________________________________________________________________________________
batch_normalization_28 (BatchNo (None, 12, 35, 128)  512         conv2d_28[0][0]                  
__________________________________________________________________________________________________
batch_normalization_6 (BatchNor (None, 48, 48, 64)   256         conv2d_6[0][0]                   
__________________________________________________________________________________________________
activation_37 (Activation)      (None, 12, 35, 128)  0           batch_normalization_28[0][0]     
__________________________________________________________________________________________________
activation_7 (Activation)       (None, 48, 48, 64)   0           batch_normalization_6[0][0]      
__________________________________________________________________________________________________
conv2d_29 (Conv2D)              (None, 12, 35, 128)  147584      activation_37[0][0]              
__________________________________________________________________________________________________
add_2 (Add)                     (None, 48, 48, 64)   0           activation_7[0][0]               
                                                                 activation_5[0][0]               
__________________________________________________________________________________________________
batch_normalization_29 (BatchNo (None, 12, 35, 128)  512         conv2d_29[0][0]                  
__________________________________________________________________________________________________
activation_8 (Activation)       (None, 48, 48, 64)   0           add_2[0][0]                      
__________________________________________________________________________________________________
activation_38 (Activation)      (None, 12, 35, 128)  0           batch_normalization_29[0][0]     
__________________________________________________________________________________________________
conv2d_7 (Conv2D)               (None, 24, 24, 128)  73856       activation_8[0][0]               
__________________________________________________________________________________________________
add_10 (Add)                    (None, 12, 35, 128)  0           activation_38[0][0]              
                                                                 activation_36[0][0]              
__________________________________________________________________________________________________
batch_normalization_7 (BatchNor (None, 24, 24, 128)  512         conv2d_7[0][0]                   
__________________________________________________________________________________________________
activation_39 (Activation)      (None, 12, 35, 128)  0           add_10[0][0]                     
__________________________________________________________________________________________________
activation_9 (Activation)       (None, 24, 24, 128)  0           batch_normalization_7[0][0]      
__________________________________________________________________________________________________
conv2d_30 (Conv2D)              (None, 12, 35, 128)  147584      activation_39[0][0]              
__________________________________________________________________________________________________
conv2d_8 (Conv2D)               (None, 24, 24, 128)  147584      activation_9[0][0]               
__________________________________________________________________________________________________
batch_normalization_30 (BatchNo (None, 12, 35, 128)  512         conv2d_30[0][0]                  
__________________________________________________________________________________________________
batch_normalization_8 (BatchNor (None, 24, 24, 128)  512         conv2d_8[0][0]                   
__________________________________________________________________________________________________
activation_40 (Activation)      (None, 12, 35, 128)  0           batch_normalization_30[0][0]     
__________________________________________________________________________________________________
activation_10 (Activation)      (None, 24, 24, 128)  0           batch_normalization_8[0][0]      
__________________________________________________________________________________________________
conv2d_31 (Conv2D)              (None, 12, 35, 128)  147584      activation_40[0][0]              
__________________________________________________________________________________________________
conv2d_9 (Conv2D)               (None, 24, 24, 128)  147584      activation_10[0][0]              
__________________________________________________________________________________________________
batch_normalization_31 (BatchNo (None, 12, 35, 128)  512         conv2d_31[0][0]                  
__________________________________________________________________________________________________
batch_normalization_9 (BatchNor (None, 24, 24, 128)  512         conv2d_9[0][0]                   
__________________________________________________________________________________________________
activation_41 (Activation)      (None, 12, 35, 128)  0           batch_normalization_31[0][0]     
__________________________________________________________________________________________________
activation_11 (Activation)      (None, 24, 24, 128)  0           batch_normalization_9[0][0]      
__________________________________________________________________________________________________
add_11 (Add)                    (None, 12, 35, 128)  0           activation_41[0][0]              
                                                                 activation_39[0][0]              
__________________________________________________________________________________________________
add_3 (Add)                     (None, 24, 24, 128)  0           activation_11[0][0]              
                                                                 activation_9[0][0]               
__________________________________________________________________________________________________
activation_42 (Activation)      (None, 12, 35, 128)  0           add_11[0][0]                     
__________________________________________________________________________________________________
activation_12 (Activation)      (None, 24, 24, 128)  0           add_3[0][0]                      
__________________________________________________________________________________________________
conv2d_32 (Conv2D)              (None, 12, 35, 128)  147584      activation_42[0][0]              
__________________________________________________________________________________________________
conv2d_10 (Conv2D)              (None, 24, 24, 128)  147584      activation_12[0][0]              
__________________________________________________________________________________________________
batch_normalization_32 (BatchNo (None, 12, 35, 128)  512         conv2d_32[0][0]                  
__________________________________________________________________________________________________
batch_normalization_10 (BatchNo (None, 24, 24, 128)  512         conv2d_10[0][0]                  
__________________________________________________________________________________________________
activation_43 (Activation)      (None, 12, 35, 128)  0           batch_normalization_32[0][0]     
__________________________________________________________________________________________________
activation_13 (Activation)      (None, 24, 24, 128)  0           batch_normalization_10[0][0]     
__________________________________________________________________________________________________
conv2d_33 (Conv2D)              (None, 12, 35, 128)  147584      activation_43[0][0]              
__________________________________________________________________________________________________
conv2d_11 (Conv2D)              (None, 24, 24, 128)  147584      activation_13[0][0]              
__________________________________________________________________________________________________
batch_normalization_33 (BatchNo (None, 12, 35, 128)  512         conv2d_33[0][0]                  
__________________________________________________________________________________________________
batch_normalization_11 (BatchNo (None, 24, 24, 128)  512         conv2d_11[0][0]                  
__________________________________________________________________________________________________
activation_44 (Activation)      (None, 12, 35, 128)  0           batch_normalization_33[0][0]     
__________________________________________________________________________________________________
activation_14 (Activation)      (None, 24, 24, 128)  0           batch_normalization_11[0][0]     
__________________________________________________________________________________________________
add_12 (Add)                    (None, 12, 35, 128)  0           activation_44[0][0]              
                                                                 activation_42[0][0]              
__________________________________________________________________________________________________
add_4 (Add)                     (None, 24, 24, 128)  0           activation_14[0][0]              
                                                                 activation_12[0][0]              
__________________________________________________________________________________________________
activation_45 (Activation)      (None, 12, 35, 128)  0           add_12[0][0]                     
__________________________________________________________________________________________________
activation_15 (Activation)      (None, 24, 24, 128)  0           add_4[0][0]                      
__________________________________________________________________________________________________
zero_padding2d_1 (ZeroPadding2D (None, 13, 35, 128)  0           activation_45[0][0]              
__________________________________________________________________________________________________
conv2d_12 (Conv2D)              (None, 24, 24, 128)  147584      activation_15[0][0]              
__________________________________________________________________________________________________
conv2d_34 (Conv2D)              (None, 13, 18, 256)  295168      zero_padding2d_1[0][0]           
__________________________________________________________________________________________________
batch_normalization_12 (BatchNo (None, 24, 24, 128)  512         conv2d_12[0][0]                  
__________________________________________________________________________________________________
batch_normalization_34 (BatchNo (None, 13, 18, 256)  1024        conv2d_34[0][0]                  
__________________________________________________________________________________________________
activation_16 (Activation)      (None, 24, 24, 128)  0           batch_normalization_12[0][0]     
__________________________________________________________________________________________________
activation_46 (Activation)      (None, 13, 18, 256)  0           batch_normalization_34[0][0]     
__________________________________________________________________________________________________
conv2d_13 (Conv2D)              (None, 24, 24, 128)  147584      activation_16[0][0]              
__________________________________________________________________________________________________
conv2d_35 (Conv2D)              (None, 13, 18, 256)  590080      activation_46[0][0]              
__________________________________________________________________________________________________
batch_normalization_13 (BatchNo (None, 24, 24, 128)  512         conv2d_13[0][0]                  
__________________________________________________________________________________________________
batch_normalization_35 (BatchNo (None, 13, 18, 256)  1024        conv2d_35[0][0]                  
__________________________________________________________________________________________________
activation_17 (Activation)      (None, 24, 24, 128)  0           batch_normalization_13[0][0]     
__________________________________________________________________________________________________
activation_47 (Activation)      (None, 13, 18, 256)  0           batch_normalization_35[0][0]     
__________________________________________________________________________________________________
add_5 (Add)                     (None, 24, 24, 128)  0           activation_17[0][0]              
                                                                 activation_15[0][0]              
__________________________________________________________________________________________________
conv2d_36 (Conv2D)              (None, 13, 18, 256)  590080      activation_47[0][0]              
__________________________________________________________________________________________________
activation_18 (Activation)      (None, 24, 24, 128)  0           add_5[0][0]                      
__________________________________________________________________________________________________
batch_normalization_36 (BatchNo (None, 13, 18, 256)  1024        conv2d_36[0][0]                  
__________________________________________________________________________________________________
conv2d_14 (Conv2D)              (None, 12, 12, 256)  295168      activation_18[0][0]              
__________________________________________________________________________________________________
activation_48 (Activation)      (None, 13, 18, 256)  0           batch_normalization_36[0][0]     
__________________________________________________________________________________________________
batch_normalization_14 (BatchNo (None, 12, 12, 256)  1024        conv2d_14[0][0]                  
__________________________________________________________________________________________________
add_13 (Add)                    (None, 13, 18, 256)  0           activation_48[0][0]              
                                                                 activation_46[0][0]              
__________________________________________________________________________________________________
activation_19 (Activation)      (None, 12, 12, 256)  0           batch_normalization_14[0][0]     
__________________________________________________________________________________________________
activation_49 (Activation)      (None, 13, 18, 256)  0           add_13[0][0]                     
__________________________________________________________________________________________________
conv2d_15 (Conv2D)              (None, 12, 12, 256)  590080      activation_19[0][0]              
__________________________________________________________________________________________________
conv2d_37 (Conv2D)              (None, 13, 18, 256)  590080      activation_49[0][0]              
__________________________________________________________________________________________________
batch_normalization_15 (BatchNo (None, 12, 12, 256)  1024        conv2d_15[0][0]                  
__________________________________________________________________________________________________
batch_normalization_37 (BatchNo (None, 13, 18, 256)  1024        conv2d_37[0][0]                  
__________________________________________________________________________________________________
activation_20 (Activation)      (None, 12, 12, 256)  0           batch_normalization_15[0][0]     
__________________________________________________________________________________________________
activation_50 (Activation)      (None, 13, 18, 256)  0           batch_normalization_37[0][0]     
__________________________________________________________________________________________________
conv2d_16 (Conv2D)              (None, 12, 12, 256)  590080      activation_20[0][0]              
__________________________________________________________________________________________________
conv2d_38 (Conv2D)              (None, 13, 18, 256)  590080      activation_50[0][0]              
__________________________________________________________________________________________________
batch_normalization_16 (BatchNo (None, 12, 12, 256)  1024        conv2d_16[0][0]                  
__________________________________________________________________________________________________
batch_normalization_38 (BatchNo (None, 13, 18, 256)  1024        conv2d_38[0][0]                  
__________________________________________________________________________________________________
activation_21 (Activation)      (None, 12, 12, 256)  0           batch_normalization_16[0][0]     
__________________________________________________________________________________________________
activation_51 (Activation)      (None, 13, 18, 256)  0           batch_normalization_38[0][0]     
__________________________________________________________________________________________________
add_6 (Add)                     (None, 12, 12, 256)  0           activation_21[0][0]              
                                                                 activation_19[0][0]              
__________________________________________________________________________________________________
add_14 (Add)                    (None, 13, 18, 256)  0           activation_51[0][0]              
                                                                 activation_49[0][0]              
__________________________________________________________________________________________________
activation_22 (Activation)      (None, 12, 12, 256)  0           add_6[0][0]                      
__________________________________________________________________________________________________
activation_52 (Activation)      (None, 13, 18, 256)  0           add_14[0][0]                     
__________________________________________________________________________________________________
conv2d_17 (Conv2D)              (None, 12, 12, 256)  590080      activation_22[0][0]              
__________________________________________________________________________________________________
conv2d_39 (Conv2D)              (None, 7, 9, 512)    1180160     activation_52[0][0]              
__________________________________________________________________________________________________
batch_normalization_17 (BatchNo (None, 12, 12, 256)  1024        conv2d_17[0][0]                  
__________________________________________________________________________________________________
batch_normalization_39 (BatchNo (None, 7, 9, 512)    2048        conv2d_39[0][0]                  
__________________________________________________________________________________________________
activation_23 (Activation)      (None, 12, 12, 256)  0           batch_normalization_17[0][0]     
__________________________________________________________________________________________________
activation_53 (Activation)      (None, 7, 9, 512)    0           batch_normalization_39[0][0]     
__________________________________________________________________________________________________
conv2d_18 (Conv2D)              (None, 12, 12, 256)  590080      activation_23[0][0]              
__________________________________________________________________________________________________
conv2d_40 (Conv2D)              (None, 7, 9, 512)    2359808     activation_53[0][0]              
__________________________________________________________________________________________________
batch_normalization_18 (BatchNo (None, 12, 12, 256)  1024        conv2d_18[0][0]                  
__________________________________________________________________________________________________
batch_normalization_40 (BatchNo (None, 7, 9, 512)    2048        conv2d_40[0][0]                  
__________________________________________________________________________________________________
activation_24 (Activation)      (None, 12, 12, 256)  0           batch_normalization_18[0][0]     
__________________________________________________________________________________________________
activation_54 (Activation)      (None, 7, 9, 512)    0           batch_normalization_40[0][0]     
__________________________________________________________________________________________________
add_7 (Add)                     (None, 12, 12, 256)  0           activation_24[0][0]              
                                                                 activation_22[0][0]              
__________________________________________________________________________________________________
conv2d_41 (Conv2D)              (None, 7, 9, 512)    2359808     activation_54[0][0]              
__________________________________________________________________________________________________
activation_25 (Activation)      (None, 12, 12, 256)  0           add_7[0][0]                      
__________________________________________________________________________________________________
batch_normalization_41 (BatchNo (None, 7, 9, 512)    2048        conv2d_41[0][0]                  
__________________________________________________________________________________________________
conv2d_19 (Conv2D)              (None, 6, 6, 512)    1180160     activation_25[0][0]              
__________________________________________________________________________________________________
activation_55 (Activation)      (None, 7, 9, 512)    0           batch_normalization_41[0][0]     
__________________________________________________________________________________________________
batch_normalization_19 (BatchNo (None, 6, 6, 512)    2048        conv2d_19[0][0]                  
__________________________________________________________________________________________________
add_15 (Add)                    (None, 7, 9, 512)    0           activation_55[0][0]              
                                                                 activation_53[0][0]              
__________________________________________________________________________________________________
activation_26 (Activation)      (None, 6, 6, 512)    0           batch_normalization_19[0][0]     
__________________________________________________________________________________________________
activation_56 (Activation)      (None, 7, 9, 512)    0           add_15[0][0]                     
__________________________________________________________________________________________________
conv2d_20 (Conv2D)              (None, 6, 6, 512)    2359808     activation_26[0][0]              
__________________________________________________________________________________________________
conv2d_42 (Conv2D)              (None, 7, 9, 512)    2359808     activation_56[0][0]              
__________________________________________________________________________________________________
batch_normalization_20 (BatchNo (None, 6, 6, 512)    2048        conv2d_20[0][0]                  
__________________________________________________________________________________________________
batch_normalization_42 (BatchNo (None, 7, 9, 512)    2048        conv2d_42[0][0]                  
__________________________________________________________________________________________________
activation_27 (Activation)      (None, 6, 6, 512)    0           batch_normalization_20[0][0]     
__________________________________________________________________________________________________
activation_57 (Activation)      (None, 7, 9, 512)    0           batch_normalization_42[0][0]     
__________________________________________________________________________________________________
conv2d_21 (Conv2D)              (None, 6, 6, 512)    2359808     activation_27[0][0]              
__________________________________________________________________________________________________
conv2d_43 (Conv2D)              (None, 7, 9, 512)    2359808     activation_57[0][0]              
__________________________________________________________________________________________________
batch_normalization_21 (BatchNo (None, 6, 6, 512)    2048        conv2d_21[0][0]                  
__________________________________________________________________________________________________
batch_normalization_43 (BatchNo (None, 7, 9, 512)    2048        conv2d_43[0][0]                  
__________________________________________________________________________________________________
activation_28 (Activation)      (None, 6, 6, 512)    0           batch_normalization_21[0][0]     
__________________________________________________________________________________________________
activation_58 (Activation)      (None, 7, 9, 512)    0           batch_normalization_43[0][0]     
__________________________________________________________________________________________________
add_8 (Add)                     (None, 6, 6, 512)    0           activation_28[0][0]              
                                                                 activation_26[0][0]              
__________________________________________________________________________________________________
add_16 (Add)                    (None, 7, 9, 512)    0           activation_58[0][0]              
                                                                 activation_56[0][0]              
__________________________________________________________________________________________________
activation_29 (Activation)      (None, 6, 6, 512)    0           add_8[0][0]                      
__________________________________________________________________________________________________
activation_59 (Activation)      (None, 7, 9, 512)    0           add_16[0][0]                     
__________________________________________________________________________________________________
conv2d_22 (Conv2D)              (None, 6, 6, 512)    2359808     activation_29[0][0]              
__________________________________________________________________________________________________
conv2d_44 (Conv2D)              (None, 4, 5, 512)    2359808     activation_59[0][0]              
__________________________________________________________________________________________________
batch_normalization_22 (BatchNo (None, 6, 6, 512)    2048        conv2d_22[0][0]                  
__________________________________________________________________________________________________
batch_normalization_44 (BatchNo (None, 4, 5, 512)    2048        conv2d_44[0][0]                  
__________________________________________________________________________________________________
activation_30 (Activation)      (None, 6, 6, 512)    0           batch_normalization_22[0][0]     
__________________________________________________________________________________________________
activation_60 (Activation)      (None, 4, 5, 512)    0           batch_normalization_44[0][0]     
__________________________________________________________________________________________________
conv2d_23 (Conv2D)              (None, 6, 6, 512)    2359808     activation_30[0][0]              
__________________________________________________________________________________________________
conv2d_45 (Conv2D)              (None, 4, 5, 512)    2359808     activation_60[0][0]              
__________________________________________________________________________________________________
batch_normalization_23 (BatchNo (None, 6, 6, 512)    2048        conv2d_23[0][0]                  
__________________________________________________________________________________________________
batch_normalization_45 (BatchNo (None, 4, 5, 512)    2048        conv2d_45[0][0]                  
__________________________________________________________________________________________________
activation_31 (Activation)      (None, 6, 6, 512)    0           batch_normalization_23[0][0]     
__________________________________________________________________________________________________
activation_61 (Activation)      (None, 4, 5, 512)    0           batch_normalization_45[0][0]     
__________________________________________________________________________________________________
add_9 (Add)                     (None, 6, 6, 512)    0           activation_31[0][0]              
                                                                 activation_29[0][0]              
__________________________________________________________________________________________________
conv2d_46 (Conv2D)              (None, 4, 5, 512)    2359808     activation_61[0][0]              
__________________________________________________________________________________________________
activation_32 (Activation)      (None, 6, 6, 512)    0           add_9[0][0]                      
__________________________________________________________________________________________________
batch_normalization_46 (BatchNo (None, 4, 5, 512)    2048        conv2d_46[0][0]                  
__________________________________________________________________________________________________
conv2d_24 (Conv2D)              (None, 3, 3, 512)    2359808     activation_32[0][0]              
__________________________________________________________________________________________________
activation_62 (Activation)      (None, 4, 5, 512)    0           batch_normalization_46[0][0]     
__________________________________________________________________________________________________
batch_normalization_24 (BatchNo (None, 3, 3, 512)    2048        conv2d_24[0][0]                  
__________________________________________________________________________________________________
add_17 (Add)                    (None, 4, 5, 512)    0           activation_62[0][0]              
                                                                 activation_60[0][0]              
__________________________________________________________________________________________________
activation_33 (Activation)      (None, 3, 3, 512)    0           batch_normalization_24[0][0]     
__________________________________________________________________________________________________
activation_63 (Activation)      (None, 4, 5, 512)    0           add_17[0][0]                     
__________________________________________________________________________________________________
conv2d_25 (Conv2D)              (None, 1, 1, 512)    2359808     activation_33[0][0]              
__________________________________________________________________________________________________
conv2d_47 (Conv2D)              (None, 1, 1, 512)    5243392     activation_63[0][0]              
__________________________________________________________________________________________________
batch_normalization_25 (BatchNo (None, 1, 1, 512)    2048        conv2d_25[0][0]                  
__________________________________________________________________________________________________
batch_normalization_47 (BatchNo (None, 1, 1, 512)    2048        conv2d_47[0][0]                  
__________________________________________________________________________________________________
activation_34 (Activation)      (None, 1, 1, 512)    0           batch_normalization_25[0][0]     
__________________________________________________________________________________________________
activation_64 (Activation)      (None, 1, 1, 512)    0           batch_normalization_47[0][0]     
__________________________________________________________________________________________________
conv2d_26 (Conv2D)              (None, 1, 1, 512)    262656      activation_34[0][0]              
__________________________________________________________________________________________________
conv2d_48 (Conv2D)              (None, 1, 1, 512)    262656      activation_64[0][0]              
__________________________________________________________________________________________________
batch_normalization_26 (BatchNo (None, 1, 1, 512)    2048        conv2d_26[0][0]                  
__________________________________________________________________________________________________
batch_normalization_48 (BatchNo (None, 1, 1, 512)    2048        conv2d_48[0][0]                  
__________________________________________________________________________________________________
activation_35 (Activation)      (None, 1, 1, 512)    0           batch_normalization_26[0][0]     
__________________________________________________________________________________________________
activation_65 (Activation)      (None, 1, 1, 512)    0           batch_normalization_48[0][0]     
__________________________________________________________________________________________________
concatenate_1 (Concatenate)     (None, 1, 1, 1024)   0           activation_35[0][0]              
                                                                 activation_65[0][0]              
__________________________________________________________________________________________________
conv2d_transpose_1 (Conv2DTrans (None, 3, 3, 512)    4719104     concatenate_1[0][0]              
__________________________________________________________________________________________________
batch_normalization_49 (BatchNo (None, 3, 3, 512)    2048        conv2d_transpose_1[0][0]         
__________________________________________________________________________________________________
activation_66 (Activation)      (None, 3, 3, 512)    0           batch_normalization_49[0][0]     
__________________________________________________________________________________________________
concatenate_2 (Concatenate)     (None, 3, 3, 1024)   0           activation_33[0][0]              
                                                                 activation_66[0][0]              
__________________________________________________________________________________________________
conv2d_transpose_2 (Conv2DTrans (None, 6, 6, 512)    4719104     concatenate_2[0][0]              
__________________________________________________________________________________________________
batch_normalization_50 (BatchNo (None, 6, 6, 512)    2048        conv2d_transpose_2[0][0]         
__________________________________________________________________________________________________
activation_67 (Activation)      (None, 6, 6, 512)    0           batch_normalization_50[0][0]     
__________________________________________________________________________________________________
conv2d_49 (Conv2D)              (None, 6, 6, 512)    2359808     activation_67[0][0]              
__________________________________________________________________________________________________
batch_normalization_51 (BatchNo (None, 6, 6, 512)    2048        conv2d_49[0][0]                  
__________________________________________________________________________________________________
activation_68 (Activation)      (None, 6, 6, 512)    0           batch_normalization_51[0][0]     
__________________________________________________________________________________________________
conv2d_50 (Conv2D)              (None, 6, 6, 512)    2359808     activation_68[0][0]              
__________________________________________________________________________________________________
batch_normalization_52 (BatchNo (None, 6, 6, 512)    2048        conv2d_50[0][0]                  
__________________________________________________________________________________________________
activation_69 (Activation)      (None, 6, 6, 512)    0           batch_normalization_52[0][0]     
__________________________________________________________________________________________________
add_18 (Add)                    (None, 6, 6, 512)    0           activation_69[0][0]              
                                                                 activation_67[0][0]              
__________________________________________________________________________________________________
activation_70 (Activation)      (None, 6, 6, 512)    0           add_18[0][0]                     
__________________________________________________________________________________________________
conv2d_51 (Conv2D)              (None, 6, 6, 512)    2359808     activation_70[0][0]              
__________________________________________________________________________________________________
batch_normalization_53 (BatchNo (None, 6, 6, 512)    2048        conv2d_51[0][0]                  
__________________________________________________________________________________________________
activation_71 (Activation)      (None, 6, 6, 512)    0           batch_normalization_53[0][0]     
__________________________________________________________________________________________________
conv2d_52 (Conv2D)              (None, 6, 6, 512)    2359808     activation_71[0][0]              
__________________________________________________________________________________________________
batch_normalization_54 (BatchNo (None, 6, 6, 512)    2048        conv2d_52[0][0]                  
__________________________________________________________________________________________________
activation_72 (Activation)      (None, 6, 6, 512)    0           batch_normalization_54[0][0]     
__________________________________________________________________________________________________
add_19 (Add)                    (None, 6, 6, 512)    0           activation_72[0][0]              
                                                                 activation_70[0][0]              
__________________________________________________________________________________________________
activation_73 (Activation)      (None, 6, 6, 512)    0           add_19[0][0]                     
__________________________________________________________________________________________________
concatenate_3 (Concatenate)     (None, 6, 6, 1024)   0           activation_32[0][0]              
                                                                 activation_73[0][0]              
__________________________________________________________________________________________________
conv2d_transpose_3 (Conv2DTrans (None, 12, 12, 256)  2359552     concatenate_3[0][0]              
__________________________________________________________________________________________________
batch_normalization_55 (BatchNo (None, 12, 12, 256)  1024        conv2d_transpose_3[0][0]         
__________________________________________________________________________________________________
activation_74 (Activation)      (None, 12, 12, 256)  0           batch_normalization_55[0][0]     
__________________________________________________________________________________________________
conv2d_53 (Conv2D)              (None, 12, 12, 256)  590080      activation_74[0][0]              
__________________________________________________________________________________________________
batch_normalization_56 (BatchNo (None, 12, 12, 256)  1024        conv2d_53[0][0]                  
__________________________________________________________________________________________________
activation_75 (Activation)      (None, 12, 12, 256)  0           batch_normalization_56[0][0]     
__________________________________________________________________________________________________
conv2d_54 (Conv2D)              (None, 12, 12, 256)  590080      activation_75[0][0]              
__________________________________________________________________________________________________
batch_normalization_57 (BatchNo (None, 12, 12, 256)  1024        conv2d_54[0][0]                  
__________________________________________________________________________________________________
activation_76 (Activation)      (None, 12, 12, 256)  0           batch_normalization_57[0][0]     
__________________________________________________________________________________________________
add_20 (Add)                    (None, 12, 12, 256)  0           activation_76[0][0]              
                                                                 activation_74[0][0]              
__________________________________________________________________________________________________
activation_77 (Activation)      (None, 12, 12, 256)  0           add_20[0][0]                     
__________________________________________________________________________________________________
conv2d_55 (Conv2D)              (None, 12, 12, 256)  590080      activation_77[0][0]              
__________________________________________________________________________________________________
batch_normalization_58 (BatchNo (None, 12, 12, 256)  1024        conv2d_55[0][0]                  
__________________________________________________________________________________________________
activation_78 (Activation)      (None, 12, 12, 256)  0           batch_normalization_58[0][0]     
__________________________________________________________________________________________________
conv2d_56 (Conv2D)              (None, 12, 12, 256)  590080      activation_78[0][0]              
__________________________________________________________________________________________________
batch_normalization_59 (BatchNo (None, 12, 12, 256)  1024        conv2d_56[0][0]                  
__________________________________________________________________________________________________
activation_79 (Activation)      (None, 12, 12, 256)  0           batch_normalization_59[0][0]     
__________________________________________________________________________________________________
add_21 (Add)                    (None, 12, 12, 256)  0           activation_79[0][0]              
                                                                 activation_77[0][0]              
__________________________________________________________________________________________________
activation_80 (Activation)      (None, 12, 12, 256)  0           add_21[0][0]                     
__________________________________________________________________________________________________
concatenate_4 (Concatenate)     (None, 12, 12, 512)  0           activation_25[0][0]              
                                                                 activation_80[0][0]              
__________________________________________________________________________________________________
conv2d_transpose_4 (Conv2DTrans (None, 24, 24, 128)  589952      concatenate_4[0][0]              
__________________________________________________________________________________________________
batch_normalization_60 (BatchNo (None, 24, 24, 128)  512         conv2d_transpose_4[0][0]         
__________________________________________________________________________________________________
activation_81 (Activation)      (None, 24, 24, 128)  0           batch_normalization_60[0][0]     
__________________________________________________________________________________________________
conv2d_57 (Conv2D)              (None, 24, 24, 128)  147584      activation_81[0][0]              
__________________________________________________________________________________________________
batch_normalization_61 (BatchNo (None, 24, 24, 128)  512         conv2d_57[0][0]                  
__________________________________________________________________________________________________
activation_82 (Activation)      (None, 24, 24, 128)  0           batch_normalization_61[0][0]     
__________________________________________________________________________________________________
conv2d_58 (Conv2D)              (None, 24, 24, 128)  147584      activation_82[0][0]              
__________________________________________________________________________________________________
batch_normalization_62 (BatchNo (None, 24, 24, 128)  512         conv2d_58[0][0]                  
__________________________________________________________________________________________________
activation_83 (Activation)      (None, 24, 24, 128)  0           batch_normalization_62[0][0]     
__________________________________________________________________________________________________
add_22 (Add)                    (None, 24, 24, 128)  0           activation_83[0][0]              
                                                                 activation_81[0][0]              
__________________________________________________________________________________________________
activation_84 (Activation)      (None, 24, 24, 128)  0           add_22[0][0]                     
__________________________________________________________________________________________________
conv2d_59 (Conv2D)              (None, 24, 24, 128)  147584      activation_84[0][0]              
__________________________________________________________________________________________________
batch_normalization_63 (BatchNo (None, 24, 24, 128)  512         conv2d_59[0][0]                  
__________________________________________________________________________________________________
activation_85 (Activation)      (None, 24, 24, 128)  0           batch_normalization_63[0][0]     
__________________________________________________________________________________________________
conv2d_60 (Conv2D)              (None, 24, 24, 128)  147584      activation_85[0][0]              
__________________________________________________________________________________________________
batch_normalization_64 (BatchNo (None, 24, 24, 128)  512         conv2d_60[0][0]                  
__________________________________________________________________________________________________
activation_86 (Activation)      (None, 24, 24, 128)  0           batch_normalization_64[0][0]     
__________________________________________________________________________________________________
add_23 (Add)                    (None, 24, 24, 128)  0           activation_86[0][0]              
                                                                 activation_84[0][0]              
__________________________________________________________________________________________________
activation_87 (Activation)      (None, 24, 24, 128)  0           add_23[0][0]                     
__________________________________________________________________________________________________
concatenate_5 (Concatenate)     (None, 24, 24, 256)  0           activation_18[0][0]              
                                                                 activation_87[0][0]              
__________________________________________________________________________________________________
conv2d_transpose_5 (Conv2DTrans (None, 48, 48, 64)   147520      concatenate_5[0][0]              
__________________________________________________________________________________________________
batch_normalization_65 (BatchNo (None, 48, 48, 64)   256         conv2d_transpose_5[0][0]         
__________________________________________________________________________________________________
activation_88 (Activation)      (None, 48, 48, 64)   0           batch_normalization_65[0][0]     
__________________________________________________________________________________________________
conv2d_61 (Conv2D)              (None, 48, 48, 64)   36928       activation_88[0][0]              
__________________________________________________________________________________________________
batch_normalization_66 (BatchNo (None, 48, 48, 64)   256         conv2d_61[0][0]                  
__________________________________________________________________________________________________
activation_89 (Activation)      (None, 48, 48, 64)   0           batch_normalization_66[0][0]     
__________________________________________________________________________________________________
conv2d_62 (Conv2D)              (None, 48, 48, 64)   36928       activation_89[0][0]              
__________________________________________________________________________________________________
batch_normalization_67 (BatchNo (None, 48, 48, 64)   256         conv2d_62[0][0]                  
__________________________________________________________________________________________________
activation_90 (Activation)      (None, 48, 48, 64)   0           batch_normalization_67[0][0]     
__________________________________________________________________________________________________
add_24 (Add)                    (None, 48, 48, 64)   0           activation_90[0][0]              
                                                                 activation_88[0][0]              
__________________________________________________________________________________________________
activation_91 (Activation)      (None, 48, 48, 64)   0           add_24[0][0]                     
__________________________________________________________________________________________________
conv2d_63 (Conv2D)              (None, 48, 48, 64)   36928       activation_91[0][0]              
__________________________________________________________________________________________________
batch_normalization_68 (BatchNo (None, 48, 48, 64)   256         conv2d_63[0][0]                  
__________________________________________________________________________________________________
activation_92 (Activation)      (None, 48, 48, 64)   0           batch_normalization_68[0][0]     
__________________________________________________________________________________________________
conv2d_64 (Conv2D)              (None, 48, 48, 64)   36928       activation_92[0][0]              
__________________________________________________________________________________________________
batch_normalization_69 (BatchNo (None, 48, 48, 64)   256         conv2d_64[0][0]                  
__________________________________________________________________________________________________
activation_93 (Activation)      (None, 48, 48, 64)   0           batch_normalization_69[0][0]     
__________________________________________________________________________________________________
add_25 (Add)                    (None, 48, 48, 64)   0           activation_93[0][0]              
                                                                 activation_91[0][0]              
__________________________________________________________________________________________________
activation_94 (Activation)      (None, 48, 48, 64)   0           add_25[0][0]                     
__________________________________________________________________________________________________
concatenate_6 (Concatenate)     (None, 48, 48, 128)  0           activation_8[0][0]               
                                                                 activation_94[0][0]              
__________________________________________________________________________________________________
conv2d_transpose_6 (Conv2DTrans (None, 96, 96, 32)   36896       concatenate_6[0][0]              
__________________________________________________________________________________________________
batch_normalization_70 (BatchNo (None, 96, 96, 32)   128         conv2d_transpose_6[0][0]         
__________________________________________________________________________________________________
activation_95 (Activation)      (None, 96, 96, 32)   0           batch_normalization_70[0][0]     
__________________________________________________________________________________________________
concatenate_7 (Concatenate)     (None, 96, 96, 64)   0           activation_1[0][0]               
                                                                 activation_95[0][0]              
__________________________________________________________________________________________________
conv2d_65 (Conv2D)              (None, 96, 96, 16)   9232        concatenate_7[0][0]              
__________________________________________________________________________________________________
batch_normalization_71 (BatchNo (None, 96, 96, 16)   64          conv2d_65[0][0]                  
__________________________________________________________________________________________________
activation_96 (Activation)      (None, 96, 96, 16)   0           batch_normalization_71[0][0]     
__________________________________________________________________________________________________
conv2d_66 (Conv2D)              (None, 96, 96, 16)   2320        activation_96[0][0]              
__________________________________________________________________________________________________
batch_normalization_72 (BatchNo (None, 96, 96, 16)   64          conv2d_66[0][0]                  
__________________________________________________________________________________________________
activation_97 (Activation)      (None, 96, 96, 16)   0           batch_normalization_72[0][0]     
__________________________________________________________________________________________________
conv2d_67 (Conv2D)              (None, 96, 96, 3)    51          activation_97[0][0]              
__________________________________________________________________________________________________
prediction (Activation)         (None, 96, 96, 3)    0           conv2d_67[0][0]                  
==================================================================================================
Total params: 71,370,579
Trainable params: 71,331,987
Non-trainable params: 38,592
__________________________________________________________________________________________________
WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/optimizers.py:790: The name tf.train.Optimizer is deprecated. Please use tf.compat.v1.train.Optimizer instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:986: The name tf.assign_add is deprecated. Please use tf.compat.v1.assign_add instead.

WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:973: The name tf.assign is deprecated. Please use tf.compat.v1.assign instead.

Epoch 1/20000000
Traceback (most recent call last):
  File "train_unet.py", line 121, in <module>
    epochs=args.epochs, verbose=1, initial_epoch=0, callbacks = callbacks_list)
  File "/usr/local/lib/python3.7/dist-packages/keras/legacy/interfaces.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/keras/engine/training.py", line 1418, in fit_generator
    initial_epoch=initial_epoch)
  File "/usr/local/lib/python3.7/dist-packages/keras/engine/training_generator.py", line 181, in fit_generator
    generator_output = next(output_generator)
  File "/usr/local/lib/python3.7/dist-packages/keras/utils/data_utils.py", line 709, in get
    six.reraise(*sys.exc_info())
  File "/usr/local/lib/python3.7/dist-packages/six.py", line 703, in reraise
    raise value
  File "/usr/local/lib/python3.7/dist-packages/keras/utils/data_utils.py", line 685, in get
    inputs = self.queue.get(block=True).get()
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/python3.7/dist-packages/keras/utils/data_utils.py", line 626, in next_sample
    return six.next(_SHARED_SEQUENCES[uid])
  File "train_unet.py", line 60, in datagen
    mfcc_batch = np.expand_dims(np.asarray(mfcc_batch), 3)
  File "<__array_function__ internals>", line 6, in expand_dims
  File "/usr/local/lib/python3.7/dist-packages/numpy/lib/shape_base.py", line 597, in expand_dims
    axis = normalize_axis_tuple(axis, out_ndim)
  File "/usr/local/lib/python3.7/dist-packages/numpy/core/numeric.py", line 1358, in normalize_axis_tuple
    axis = tuple([normalize_axis_index(ax, ndim, argname) for ax in axis])
  File "/usr/local/lib/python3.7/dist-packages/numpy/core/numeric.py", line 1358, in <listcomp>
    axis = tuple([normalize_axis_index(ax, ndim, argname) for ax in axis])
numpy.AxisError: axis 3 is out of bounds for array of dimension 2

In [ ]:
!python train.py --data_root /content/LipGAN/preprocessed_files

Using TensorFlow backend.
Will be training on 0 images






2021-04-28 05:43:53.572821: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2021-04-28 05:43:53.577521: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2000165000 Hz
2021-04-28 05:43:53.577831: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55652e4c4e00 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-04-28 05:43:53.577873: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version








In [ ]:
!python train_unet.py --data_root /content/LipGAN/preprocessed_files

Using TensorFlow backend.
Instructions for updating:
non-resource variables are not supported in the long term
Will be training on 0 images






2021-04-28 05:05:19.220832: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2021-04-28 05:05:19.225304: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2000165000 Hz
2021-04-28 05:05:19.225604: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x562e5bcf8e00 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2021-04-28 05:05:19.225640: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version





__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
input_face (InputLayer)         

In [ ]:
!git clone https://github.com/Rudrabha/LipGAN.git

Cloning into 'LipGAN'...
remote: Enumerating objects: 202, done.
remote: Counting objects: 100% (202/202), done.
remote: Compressing objects: 100% (145/145), done.
remote: Total 202 (delta 117), reused 101 (delta 51), pack-reused 0
Receiving objects: 100% (202/202), 492.95 KiB | 2.77 MiB/s, done.
Resolving deltas: 100% (117/117), done.


In [ ]:
cp -r /content/

### Preprocess the dataset
We use Python [Librosa](https://librosa.github.io/librosa/) library to save melspectrogram features and perform face detection using dlib.  

```bash
# Please copy the appropriate LRS2 split's filelist.txt to the filelists/ folder. Example below is shown for LRS2. 
python preprocess.py --split [train|pretrain|val] --videos_data_root mvlrs_v1/ --final_data_root 


### More options while preprocessing (like number of workers, image size etc.)
python preprocess.py --help
```
### `<folder_to_store_preprocessed_files>` => preprocessed_files 
### `<downloaded part files>` => part_files_DELETE (can be deleted)
### `<downloaded data split files>` ==> filelists

In [ ]:
ls 

audio_hparams.py    generator.py  part_files_DELETE/   README.md
audio.py            License.md    preprocessed_files/  requirements.txt
batch_inference.py  logs/         preprocess.py        results/
discriminator.py    lrs2_v1.tar   __pycache__/         train.py
filelists/          mvlrs_v1/     readme.gdoc          train_unet.py


In [ ]:
!wc -l filelists/*.txt

  96318 filelists/pretrain.txt
   1243 filelists/test.txt
  45839 filelists/train.txt
   1082 filelists/val.txt
 144482 total


In [ ]:
pip install tensorflow==1.14.0 tensorflow-gpu==1.14.0

     |████████████████████████████████| 109.3MB 62kB/s 
     |████████████████████████████████| 377.1MB 39kB/s 
     |████████████████████████████████| 3.2MB 42.0MB/s 
     |████████████████████████████████| 491kB 43.6MB/s 
     |████████████████████████████████| 51kB 5.1MB/s 
  Found existing installation: tensorboard 2.4.1
    Uninstalling tensorboard-2.4.1:
      Successfully uninstalled tensorboard-2.4.1
  Found existing installation: tensorflow-estimator 2.4.0
    Uninstalling tensorflow-estimator-2.4.0:
      Successfully uninstalled tensorflow-estimator-2.4.0
  Found existing installation: tensorflow 2.4.1
    Uninstalling tensorflow-2.4.1:
      Successfully uninstalled tensorflow-2.4.1


PREPROCESS DATA

In [ ]:
cd '/content/drive/My Drive/Not_Synced'

/content/drive/My Drive/Not_Synced


In [ ]:
pwd

'/content'

In [ ]:
#cp -ri '/content/drive/My Drive/Not_Synced/lrs2_v1.tar'  /content/

^C


In [ ]:
#!tar -xvf lrs2_v1.tar

In [ ]:
ls

drive/  lrs2_v1.tar  sample_data/


In [ ]:
!tar -xvf '/content/drive/My Drive/Not_Synced/lrs2_v1.tar' 

Streaming output truncated to the last 5000 lines.
mvlrs_v1/pretrain/6290125634335003107/00004.txt
mvlrs_v1/pretrain/6290125634335003107/00008.txt
mvlrs_v1/pretrain/6290125634335003107/00027.txt
mvlrs_v1/pretrain/6290125634335003107/00002.mp4
mvlrs_v1/pretrain/6290125634335003107/00026.txt
mvlrs_v1/pretrain/6290125634335003107/00011.mp4
mvlrs_v1/pretrain/6290125634335003107/00018.txt
mvlrs_v1/pretrain/6290125634335003107/00004.mp4
mvlrs_v1/pretrain/6290125634335003107/00025.mp4
mvlrs_v1/pretrain/6290125634335003107/00022.txt
mvlrs_v1/pretrain/6290125634335003107/00020.mp4
mvlrs_v1/pretrain/6290125634335003107/00019.txt
mvlrs_v1/pretrain/6290125634335003107/00017.mp4
mvlrs_v1/pretrain/6290125634335003107/00027.mp4
mvlrs_v1/pretrain/6290125634335003107/00010.mp4
mvlrs_v1/pretrain/6290125634335003107/00028.txt
mvlrs_v1/pretrain/6290125634335003107/00026.mp4
mvlrs_v1/pretrain/6290125634335003107/00022.mp4
mvlrs_v1/pretrain/6290125634335003107/00013.mp4
mvlrs_v1/pretrain/6290125634335003107

In [ ]:
cp -r /content/drive/MyDrive/Not_Synced/*.py /content

In [ ]:
cp -r /content/drive/MyDrive/Not_Synced/filelists /content

In [ ]:
!sudo apt-get install ffmpeg

Reading package lists... Done
Building dependency tree       
Reading state information... Done
ffmpeg is already the newest version (7:3.4.8-0ubuntu0.2).
0 upgraded, 0 newly installed, 0 to remove and 30 not upgraded.


In [ ]:
!python preprocess.py --split train --videos_data_root mvlrs_v1/ --final_data_root preprocessed_files/ --num_workers 1
#split = [train|pretrain|val] 

/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or 

In [ ]:
cd /content/preprocessed_files/

'/content'

In [ ]:
!tar -czvf processed_folder.tar.gz /content/preprocessed_files/

Streaming output truncated to the last 5000 lines.
/content/preprocessed_files/main/6117976902660304352/00016/18.jpg
/content/preprocessed_files/main/6117976902660304352/00016/0.jpg
/content/preprocessed_files/main/6117976902660304352/00007/
/content/preprocessed_files/main/6117976902660304352/00007/mels.npz
/content/preprocessed_files/main/6117976902660304352/00007/15.jpg
/content/preprocessed_files/main/6117976902660304352/00007/25.jpg
/content/preprocessed_files/main/6117976902660304352/00007/14.jpg
/content/preprocessed_files/main/6117976902660304352/00007/11.jpg
/content/preprocessed_files/main/6117976902660304352/00007/20.jpg
/content/preprocessed_files/main/6117976902660304352/00007/19.jpg
/content/preprocessed_files/main/6117976902660304352/00007/audio.wav
/content/preprocessed_files/main/6117976902660304352/00007/10.jpg
/content/preprocessed_files/main/6117976902660304352/00007/17.jpg
/content/preprocessed_files/main/6117976902660304352/00007/9.jpg
/content/preprocessed_files/

In [ ]:
ls

audio_hparams.py    drive/        preprocessed_files/      sample_data/
audio.py            filelists/    preprocess.py            train.py
batch_inference.py  generator.py  processed_folder.tar.gz  train_unet.py
discriminator.py    mvlrs_v1/     __pycache__/


In [ ]:
cp -r  /content/processed_folder.tar.gz /content/drive/MyDrive/Not_Synced/

In [ ]:
ls /content/drive/MyDrive/Not_Synced

audio_hparams.py    logs/                    readme.gdoc
audio.py            lrs2_v1.tar              README.md
batch_inference.py  part_files_DELETE/       requirements.txt
discriminator.py    preprocessed_files/      results/
filelists/          preprocess.py            train.py
generator.py        processed_folder.tar.gz  train_unet.py
License.md          __pycache__/


In [ ]:
#@title Input directory and unit of measurement

# Credit:
# https://stackoverflow.com/a/1392549
import os
import sys
def get_size(start_path = '.', unit='MB'):
    total_size = 0
    for dirpath, dirnames, filenames in os.walk(start_path):
        for f in filenames:
            fp = os.path.join(dirpath, f)
            # skip if it is symbolic link
            if not os.path.islink(fp):
                total_size += os.path.getsize(fp)

    if unit == 'MB':
      total_size *= 1e-6
    elif unit == 'GB':
      total_size *= 1e-9
    elif unit == 'TB':
      total_size *= 1e-12

    return total_size

target_dir = '/content/drive/MyDrive/Not_Synced/processed_folder.tar.gz' #@param {type: 'string'}
output_unit = 'GB' #@param ['MB', 'GB', 'TB']

dir_size = round(get_size(target_dir, output_unit), 2)
print('Target directory:', target_dir)
print('Directory size is', dir_size, output_unit)

Target directory: /content/drive/MyDrive/Not_Synced/processed_folder.tar.gz
Directory size is 0.0 GB


In [ ]:
os.path.getsize('/content/drive/MyDrive/Not_Synced/processed_folder.tar.gz')* 1e-9

12.881889882000001

In [ ]:
cd /content/drive/MyDrive/Not_Synced/

/content/drive/MyDrive/Not_Synced


In [ ]:
ls

audio_hparams.py    logs/                    readme.gdoc
audio.py            lrs2_v1.tar              README.md
batch_inference.py  part_files_DELETE/       requirements.txt
discriminator.py    preprocessed_files/      results/
filelists/          preprocess.py            train.py
generator.py        processed_folder.tar.gz  train_unet.py
License.md          __pycache__/


In [ ]:
rm -rf preprocessed_files/

In [ ]:
cd /content/drive/My Drive/Not_Synced

/content/drive/My Drive/Not_Synced


In [ ]:
ls

audio_hparams.py    logs/                    README.md
audio.py            lrs2_v1.tar              requirements.txt
batch_inference.py  part_files_DELETE/       results/
discriminator.py    preprocess.py            train.py
filelists/          processed_folder.tar.gz  train_unet.py
generator.py        __pycache__/
License.md          readme.gdoc


In [ ]:
!tar -xvf '/content/drive/My Drive/Not_Synced/processed_folder.tar.gz' -C '/content/drive/My Drive/Not_Synced'

Streaming output truncated to the last 5000 lines.
content/preprocessed_files/main/6117976902660304352/00016/18.jpg
content/preprocessed_files/main/6117976902660304352/00016/0.jpg
content/preprocessed_files/main/6117976902660304352/00007/
content/preprocessed_files/main/6117976902660304352/00007/mels.npz
content/preprocessed_files/main/6117976902660304352/00007/15.jpg
content/preprocessed_files/main/6117976902660304352/00007/25.jpg
content/preprocessed_files/main/6117976902660304352/00007/14.jpg
content/preprocessed_files/main/6117976902660304352/00007/11.jpg
content/preprocessed_files/main/6117976902660304352/00007/20.jpg
content/preprocessed_files/main/6117976902660304352/00007/19.jpg
content/preprocessed_files/main/6117976902660304352/00007/audio.wav
content/preprocessed_files/main/6117976902660304352/00007/10.jpg
content/preprocessed_files/main/6117976902660304352/00007/17.jpg
content/preprocessed_files/main/6117976902660304352/00007/9.jpg
content/preprocessed_files/main/6117976902

In [ ]:
ls /content/

audio_hparams.py    drive/        preprocessed_files/      sample_data/
audio.py            filelists/    preprocess.py            train.py
batch_inference.py  generator.py  processed_folder.tar.gz  train_unet.py
discriminator.py    mvlrs_v1/     __pycache__/


In [ ]:
ls

audio_hparams.py    logs/                    README.md
audio.py            lrs2_v1.tar              requirements.txt
batch_inference.py  part_files_DELETE/       results/
discriminator.py    preprocess.py            train.py
filelists/          processed_folder.tar.gz  train_unet.py
generator.py        __pycache__/
License.md          readme.gdoc


Python3.6

In [ ]:
!apt-get install python3.6

In [ ]:
!apt-get install ffmpeg

In [ ]:
%cd /content/drive/MyDrive/766_processed_lip_GAN

In [ ]:
#@title
%tensorflow_version 1.x
import os
from os.path import exists, join, basename, splitext

git_repo_url = 'https://github.com/Rudrabha/LipGAN.git'
project_name = splitext(basename(git_repo_url))[0]
if not exists(project_name):
  # clone and install dependencies
  #!git clone -q --depth 1 -b fully_pythonic {git_repo_url}
  !pip install -q gdown youtube-dl
  !pip install -q git+https://www.github.com/keras-team/keras-contrib.git

import sys
sys.path.append(project_name)
import matplotlib.pyplot as plt
from IPython.display import Audio

In [ ]:
cd LipGAN/

In [ ]:
!pip install -r requirements.txt

In [ ]:
pip install git+https://www.github.com/keras-team/keras-contrib.git

In [ ]:
!python train.py --help

In [ ]:
!python train.py --data_root './data_root' --epochs 200